### PR TITLE
Implement SpotX as Ad provider

### PR DESCRIPTION
**Pull request by KenneyE**
- Implements SpotX as an ad provider. 
- Allows a publisher to use `type=spotx` when using the `amp-ad` tag.

### CODE REVIEW

File: examples/ads.amp.html [hunk 1]
```diff
@@ -1085,6 +1086,14 @@ <h2 id="sovrn">SOVRN</h2>
       data-z="393900"><!-- this ID is only whitelisted for localhost:8000 and http://amphtml-nightly.herokuapp.com-->
   </amp-ad>
 
+  <h2 id="spotx">SpotX</h2>
+  <amp-ad width="300" height="250"
+          type="spotx"
+          data-spotx_channel_id="85394"
```
Comment (lannka) [inline examples/ads.amp.html]: you can do `data-spotx-channel-id`, and read as `dataSpotxChannelId`

File: examples/ads.amp.html [hunk 2]
```diff
@@ -1085,6 +1086,14 @@ <h2 id="sovrn">SOVRN</h2>
...
+          data-spotx_autoplay="1"
+  >
```
Comment (lannka) [inline examples/ads.amp.html]: move to previous line

File: examples/ads.amp.html (hunk 1 above)
Comment (KenneyE) [inline examples/ads.amp.html]: We are hoping to match the formatting here to that used in another of our popular integrations making it easier for publishers to adopt this. In that integration, options are passed to our script using the `data-spotx_<option>` format.

File: examples/ads.amp.html (hunk 2 above)
Comment (KenneyE) [inline examples/ads.amp.html]: Done.

File: examples/ads.amp.html (hunk 1 above)
Comment (lannka) [inline examples/ads.amp.html]: what other integration are you talking about?

**Comment by lannka** [comment]
Please follow the googlebot instruction to sign CLA

File: examples/ads.amp.html (hunk 1 above)
Comment (KenneyE) [inline examples/ads.amp.html]: One of SpotX's standard ad players. Most of our publishers that will adopt the AMP integration will be coming from another ad player of ours that uses the format described above.

**Comment by KenneyE** [comment]
I believe the CLA has been signed by the Authorized Signer for the organization "SpotX"

File: ads/spotx.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+  }
+
+  global['spotx_ad_done_function'] = function(spotxAdFound) {
+    if (!spotxAdFound) {
+      global.context.noContentAvailable();
+    }
+  };
+
+  script.onload = global.context.renderStart;
```
Comment (lannka) [inline ads/spotx.js]: are you able to call `renderStart` at a better time inside the script?

File: examples/ads.amp.html [hunk 4]
```diff
@@ -1103,6 +1104,13 @@ <h2 class="broken">PulsePoint Header Bidding 300x250</h2>
       data-z="393900"><!-- this ID is only whitelisted for localhost:8000 and http://amphtml-nightly.herokuapp.com-->
   </amp-ad>
 
+  <h2>SpotX</h2>
+  <amp-ad width="300" height="250"
+          type="spotx"
```
Comment (lannka) [inline examples/ads.amp.html]: nit: please follow the indentation of other examples

**Comment by lannka** [comment]
@KenneyE to pass the CLA check, can you make sure your email is in the signed group?

File: ads/spotx.js (hunk 3 above)
Comment (KenneyE) [inline ads/spotx.js]: Unfortunately not. Would it be preferred to not use `renderStart` at all if it isn't directly connected to the ad initialization?

File: examples/ads.amp.html (hunk 4 above)
Comment (KenneyE) [inline examples/ads.amp.html]: Done

**Comment by lannka** [comment]
@KenneyE I'm curious what did you do to pass the CLA bot? Was it a commit with a different email? Or did you add your email to the signed group?

That will help me understand the bot better. Thanks!

**Comment by KenneyE** [comment]
@lannka It was pushed with the same email, but I think it might have been the first new commit since getting the CLA signed.

File: ads/spotx.js (hunk 3 above)
Comment (lannka) [inline ads/spotx.js]: is it because the script is not in your control or other reason? calling renderStart at the right time gives much better user experience, so we push hard to get it right :-)

Comment (KenneyE) [inline ads/spotx.js]: Understandable. The the script is within our control but doesn't currently indicate when rendering has started. It could be a bit before we are able to add and release that feature. Maybe it is best we don't implement `renderStart` right now, and we will work to add that ability to our script. When that is complete we will submit another pull request. Is that an acceptable path forward?

Comment (lannka) [inline ads/spotx.js]: no need to revert the change, since this is still better than nothing.
can you put a TODO and I'll be happy to merge the PR. thanks!

Comment (lannka) [inline ads/spotx.js]: for TODO, we follow the format: `TODO(yourname, #1234)`

**Comment by KenneyE** [comment]
Thanks for the guidance @lannka!
